app/src/main.js

```diff
@@ -120,8 +125,32 @@ if (appArgs.crashReporter) {
 }
 
 app.on('ready', () => {
-  mainWindow = createMainWindow(appArgs, app.quit, setDockBadge);
-  createTrayIcon(appArgs, mainWindow);
+  onlineStatusWindow = new BrowserWindow({
+    width: 300,
+    height: 200,
+    frame: false,
+    resizable: false,
+    show: false,
+    backgroundColor: 'white',
+    title: appArgs.title,
+    icon: getAppIcon(),
+  });
+  onlineStatusWindow.loadURL(`file://${__dirname}/static/online-status.html`);
+});
+
+ipcMain.on('online-status-changed', (event, status) => {
+  if (status === 'online') {
+    onlineStatusWindow.hide();
```

Comment:
Hmmmmm how does the the experience feel? Isn't it a lot of movement / jumping windows? Rather than a separate `BrowserWindow`, did you experiment with using the current one if offline, then replacing the contents with the nativefied website once offline?